profitbricksapi/profitbricksapi.php

```diff
@@ -0,0 +1,259 @@
+ * @file
+ * @version 0.1
+ * @copyright 2017 CN-Consult GmbH
+ * @author Jens Stahl <jens.stahl@cn-consult.eu>
+ *
+ * License: Please check the LICENSE file for more information.
+ */
+
+namespace ProfitBricksApi;
+
+use Exception;
+
+/**
+ * Class ProfitBricksApi connects to the official ProfitBricksApi and returns easy usable objects.
+ */
+class ProfitBricksApi
+{
+    private $user = "";
+    private $password = "";
+    const profitBricksSnapshotApi = "https://api.profitbricks.com/cloudapi/v3/snapshots?depth=1";
+    const profitBricksDataCenterApi = "https://api.profitbricks.com/cloudapi/v3/datacenters?depth=2";
+    const profitBricksServerApi = "https://api.profitbricks.com/cloudapi/v3/datacenters/[data-center-id]/servers?depth=2";
+    const profitBricksDiskApi = "https://api.profitbricks.com/cloudapi/v3/datacenters/[data-center-id]/servers/[server-id]/volumes?depth=1";
+    const profitBricksDeleteSnapshot = "https://api.profitbricks.com/cloudapi/v3/snapshots/[snapshot-id]";
+    const profitBricksCreateSnapshot = "https://api.profitbricks.com/cloudapi/v3/datacenters/[data-center-id]/volumes/[volume-id]/create-snapshot";
+
+    public function __construct()
+    {
+    }
+
+    /**
+     * Reads all information about snapshots from ProfitBricks into usable objects.
+     *
+     * @return Snapshot[]|bool
+     */
+    public function snapshots()
+    {
+        $snapshots = false;
+        $profitBricksSnapshots = $this->readFromProfitBricks(self::profitBricksSnapshotApi);
+        if ($profitBricksSnapshots!==false)
+        {
+            $snapshots = array();
+            foreach ($profitBricksSnapshots as $snapshot)
+            {
+                $snapshots[$snapshot->id] = new Snapshot($snapshot);
+            }
+        }
+        return $snapshots;
+    }
+
+    /**
+     * Reads all information about DataCenters from ProfitBricks into usable objects.
+     *
+     * @return DataCenter[]|bool
+     */
+    public function dataCenters()
+    {
+        $dataCenters = false;
+        $profitBricksDataCenters = $this->readFromProfitBricks(self::profitBricksDataCenterApi);
+        if ($profitBricksDataCenters!==false)
+        {
+            $dataCenters = array();
+            foreach ($profitBricksDataCenters as $dataCenter)
+            {
+                $dataCenters[$dataCenter->id] = new DataCenter($dataCenter);
+            }
+        }
+        return $dataCenters;
+    }
+
+     /**
+     * Reads all information about virtual machines/servers from ProfitBricks into usable objects.
+     *
+     * @param DataCenter $_dataCenter determines the data center for which the virtual machines should be searched.
+     * @return VirtualMachine[]|bool
+     */
+    public function virtualMachines(DataCenter $_dataCenter)
+    {
+        $virtualMachines = false;
+        $api = str_replace("[data-center-id]", $_dataCenter->id, self::profitBricksServerApi);
+        foreach ($this->readFromProfitBricks($api) as $virtualMachine)
```

Comment:
Theoretically, it would need here this IF-check for the case that nothing usable gets returned.